src/language-js/printer-estree.js

```diff
@@ -4638,8 +4638,31 @@ function printTypeParameters(path, options, print, paramsKey) {
           n[paramsKey][0].type !== "TSIndexedAccessType" &&
           n[paramsKey][0].type !== "TSArrayType")));
 
+  function printDanglingCommentsForInline(n) {
+    if (!hasDanglingComments(n)) {
+      return "";
+    }
+    const hasOnlyBlockComments = n.comments.every(
+      handleComments.isBlockComment
+    );
+    const printed = comments.printDanglingComments(
+      path,
+      options,
+      /* sameIndent */ hasOnlyBlockComments
+    );
+    if (hasOnlyBlockComments) {
+      return printed;
+    }
+    return concat([printed, line]);
```

Comment:
```suggestion
    return concat([printed, hardline]);
```
It never should be printed as a space.